examples/plot_sleep_staging_eldele2021.py

```diff
@@ -0,0 +1,376 @@
+"""
+Sleep staging on the Sleep Physionet dataset
+============================================
+
+This tutorial shows how to train and test a sleep staging neural network with
+Braindecode. We use the attention-based model from [1]_ with the time distributed approach of [2]_
+to learn on sequences of EEG windows using the openly accessible Sleep Physionet dataset [3]_ [4]_.
+
+References
+----------
+.. [1] E. Eldele et al., "An Attention-Based Deep Learning Approach for Sleep Stage
+        Classification With Single-Channel EEG," in IEEE Transactions on Neural Systems and
+        Rehabilitation Engineering, vol. 29, pp. 809-818, 2021, doi: 10.1109/TNSRE.2021.3076234.
+
+.. [2] Chambon, S., Galtier, M., Arnal, P., Wainrib, G. and Gramfort, A.
+      (2018)A Deep Learning Architecture for Temporal Sleep Stage
+      Classification Using Multivariate and Multimodal Time Series.
+      IEEE Trans. on Neural Systems and Rehabilitation Engineering 26:
+      (758-769)
+
+.. [3] B Kemp, AH Zwinderman, B Tuk, HAC Kamphuisen, JJL Oberyé. Analysis of
+       a sleep-dependent neuronal feedback loop: the slow-wave
+       microcontinuity of the EEG. IEEE-BME 47(9):1185-1194 (2000).
+
+.. [4] Goldberger AL, Amaral LAN, Glass L, Hausdorff JM, Ivanov PCh,
+       Mark RG, Mietus JE, Moody GB, Peng C-K, Stanley HE. (2000)
+       PhysioBank, PhysioToolkit, and PhysioNet: Components of a New
+       Research Resource for Complex Physiologic Signals.
+       Circulation 101(23):e215-e220
+"""
+# Authors: Divyesh Narayanan <divyesh.narayanan@gmail.com>
+#
+# License: BSD (3-clause)
+
+
+######################################################################
+# Loading and preprocessing the dataset
+# -------------------------------------
+#
+# Loading
+# ~~~~~~~
+#
+# First, we load the data using the
+# :class:`braindecode.datasets.sleep_physionet.SleepPhysionet` class. We load
+# two recordings from two different individuals: we will use the first one to
+# train our network and the second one to evaluate performance (as in the `MNE`_
+# sleep staging example).
+#
+# .. _MNE: https://mne.tools/stable/auto_tutorials/sample-datasets/plot_sleep.html
+#
+
+from numbers import Integral
+from braindecode.datasets.sleep_physionet import SleepPhysionet
+
+dataset = SleepPhysionet(
+    subject_ids=[0, 1], recording_ids=[2], crop_wake_mins=30)
+
+######################################################################
+# Preprocessing
+# ~~~~~~~~~~~~~
+#
+# Next, we preprocess the raw data. We convert the data to microvolts and apply
+# a lowpass filter. We omit the downsampling step of [1]_ as the Sleep
+# Physionet data is already sampled at a lower 100 Hz.
+
+from braindecode.preprocessing.preprocess import preprocess, Preprocessor, scale
```

Comment:
@robintibor @gemeinl this is not nice. Could we fix this so we read something like
```suggestion
from braindecode.preprocessing import preprocess, Preprocessor, scale
```